solr/solrj/src/java/org/apache/solr/client/solrj/SolrRequest.java

```diff
@@ -130,9 +130,14 @@ public String getBasicAuthPassword() {
   // ---------------------------------------------------------
   // ---------------------------------------------------------
 
-  public SolrRequest(METHOD m, String path) {
+  public SolrRequest(METHOD m, String path, SolrRequestType defaultType) {
     this.method = m;
     this.path = path;
+    this.defaultType = defaultType;
+  }
+
+  public SolrRequest(METHOD m, String path) {
```

Comment:
The request type here would then be subtle/hidden at the caller site.  I think the caller of SolrRequest's constructor should be explicit about this.